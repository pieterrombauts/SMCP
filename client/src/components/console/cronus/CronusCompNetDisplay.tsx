import React from 'react';
import styled from 'styled-components';
import { cronusCompNetDisplay } from '../customTypes'

interface CronusCNDProps {
  className?: string;
  values: cronusCompNetDisplay;
};

const UCronusCompNetDisplay: React.FC<CronusCNDProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_1.png'></image>
        <rect x="1055" y="180" rx="5" ry="5" width="200" height="50" style={{fill:"#000000"}} />
        <rect x="1110" y="230" rx="5" ry="5" width="100" height="50" style={{fill:"#000000"}} />
        <rect x="1110" y="250" rx="5" ry="5" width="100" height="50" style={{fill:"#000000"}} />
        <text x="1080" y="217" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.compNet.time}</text>
        <text x="1130" y="257" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.compNet.standardCmds}</text>
        <text x="1100" y="295" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.compNet.loadCmds}</text>
      </svg>  
    </div>
  )
};

const CronusCompNetDisplay = styled(UCronusCompNetDisplay)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusCompNetDisplay;