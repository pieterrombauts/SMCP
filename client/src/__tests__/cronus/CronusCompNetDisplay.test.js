import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import CronusCompNetDisplay from '../../components/console/cronus/CronusCompNetDisplay'
import { shallow, mount, render} from 'enzyme';

const defaultValues = {
  compNet: {
    time: '',
    standardCmds: 10466,
    loadCmds: 11248
  }
}

describe("Cronus computer and network display", () => {
  it('renders correctly', () => {
    const wrapper = mount(<CronusCompNetDisplay values={defaultValues} time={'2019-11-19T15:00:00'} />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})