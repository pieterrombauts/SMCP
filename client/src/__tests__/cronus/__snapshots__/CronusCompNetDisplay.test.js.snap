// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cronus computer and network display renders correctly 1`] = `
.c0 svg {
  width: 100%;
  height: 100%;
}

<Styled(UCronusCompNetDisplay)
  time="2019-11-19T15:00:00"
  values={
    Object {
      "compNet": Object {
        "loadCmds": 11248,
        "standardCmds": 10466,
        "time": "",
      },
    }
  }
>
  <UCronusCompNetDisplay
    className="c0"
    time="2019-11-19T15:00:00"
    values={
      Object {
        "compNet": Object {
          "loadCmds": 11248,
          "standardCmds": 10466,
          "time": "",
        },
      }
    }
  >
    <div
      className="c0"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 1300 990"
      >
        <image
          height={990}
          href="/media/Cronus_1.png"
          width={1300}
          x={0}
          y={0}
        />
        <rect
          height="50"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#000000",
            }
          }
          width="200"
          x="1055"
          y="180"
        />
        <rect
          height="50"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#000000",
            }
          }
          width="100"
          x="1110"
          y="230"
        />
        <rect
          height="50"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#000000",
            }
          }
          width="100"
          x="1110"
          y="250"
        />
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="1080"
          y="217"
        >
          323/15:00
           GMT
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="1130"
          y="257"
        >
          10466
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="1100"
          y="295"
        >
          11248
        </text>
      </svg>
    </div>
  </UCronusCompNetDisplay>
</Styled(UCronusCompNetDisplay)>
`;
